doc/source/dev/contributor/contributor_toc.rst

```diff
@@ -28,6 +28,7 @@ Development environment
 - :ref:`building` may have some helpful hints if you need to deviate from the guides above
 - :ref:`recommended-development-setup` includes additional notes about the development setup; all of this information is contained elsewhere, but it is retained as a legacy document
 - :ref:`meson` for how to use the Meson build system (experimental, it will replace the ``distutils``-based build in the future)
+- :ref:`meson-advanced` presents advanced build options with meson
```

Comment:
As @rgommers said in https://github.com/scipy/scipy/pull/15947#issuecomment-1091226523, the TOC is already confusing for new users. I would say instead of creating a new "meson-advanced" file, it would be better to create a new "Advanced" section in the "meson" file itself. This makes the TOC short and condenses similar information in one file. What do you think @Smit-create @rgommers?

Reply:
I suggested a separate file, because I want to separate what every contributor needs from what only a handful of people will need. However I agree the ToC is confusing. I think the solution is simply not to put this description in this ToC, but only link it from the main Meson build page.